### ISSUE TITLE
Crash with OOM?

### ISSUE THREAD

**Issue by kennedyshead** [exxamalte/python-aio-georss-client#29]
Server have 12G memory to spare so I really dont get this?
```
Traceback (most recent call last):
  File "/home/knutas/.local/lib/python3.9/site-packages/homeassistant/components/gdacs/__init__.py", line 154, in async_update
    await self._feed_manager.update()
  File "/home/knutas/.hass/deps/lib/python3.9/site-packages/aio_georss_client/feed_manager.py", line 46, in update
    status, feed_entries = await self._feed.update()
  File "/home/knutas/.hass/deps/lib/python3.9/site-packages/aio_georss_client/feed.py", line 77, in update
    status, rss_data = await self._fetch()
  File "/home/knutas/.hass/deps/lib/python3.9/site-packages/aio_georss_client/feed.py", line 114, in _fetch
    feed_data = parser.parse(text)
  File "/home/knutas/.hass/deps/lib/python3.9/site-packages/aio_georss_client/xml_parser/__init__.py", line 102, in parse
    parsed_dict = xmltodict.parse(
  File "/home/knutas/.hass/deps/lib/python3.9/site-packages/xmltodict.py", line 327, in parse
    parser.Parse(xml_input, True)
xml.parsers.expat.ExpatError: out of memory: line 1, column 0
```

**Comment by kennedyshead** [comment]
https://github.com/Ymagis/ClairMeta/issues/212

**Comment by kennedyshead** [comment]
Fast fix for now would be running `apt-get install libexpat1=2.2.9-1build1 libexpat1-dev=2.2.9-1build1`, The bug originates from libexpat

**Comment by exxamalte** [comment]
Thanks for the analysis.
I was just able to reproduce this on Ubuntu 21.10 which has libexpat1 version 2.4.1-2ubuntu0.1 installed.

```
$ python3 -c 'import xmltodict; print(xmltodict.parse("<?xml version=\"1.0\" encoding=\"UTF-8\"?><rss xmlns:geo=\"http://www.w3.org/2003/01/geo/wgs84_pos#\" version=\"2.0\"><channel><item><title>Title 1</title></item></channel></rss>", process_namespaces=True))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/malte/.local/lib/python3.9/site-packages/xmltodict.py", line 327, in parse
    parser.Parse(xml_input, True)
xml.parsers.expat.ExpatError: out of memory: line 1, column 0
```

And, after adding `namespace_separator=" "`:
```
$ python3 -c 'import xmltodict; print(xmltodict.parse("<?xml version=\"1.0\" encoding=\"UTF-8\"?><rss xmlns:geo=\"http://www.w3.org/2003/01/geo/wgs84_pos#\" version=\"2.0\"><channel><item><title>Title 1</title></item></channel></rss>", process_namespaces=True, namespace_separator=" "))'
OrderedDict([('rss', OrderedDict([('@version', '2.0'), ('@xmlns', OrderedDict([('geo', 'http://www.w3.org/2003/01/geo/wgs84_pos#')])), ('channel', OrderedDict([('item', OrderedDict([('title', 'Title 1')]))]))]))])
```

**Comment by exxamalte** [comment]
It seems that a new change in the libexpat library will fix this issue, planned for version 2.4.7: https://github.com/libexpat/libexpat/pull/577

**Comment by kennedyshead** [comment]
Closing, seems to be solved
